spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/AvroDataTestBase.java

```diff
@@ -603,4 +603,57 @@ public void testRowLineage() throws Exception {
             record.copy(Map.of("id", 4L, "data", "d", "_row_id", 1_001L)),
             record.copy(Map.of("id", 5L, "data", "e"))));
   }
+
+  @Test
+  public void testUnknownTopLevel() throws IOException {
```

Comment:
This should be covered by the existing `testTypeSchema` test: https://github.com/apache/iceberg/blob/main/spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/AvroDataTestBase.java#L164-L190

I think it's also good to test the map and list restrictions and I like that this has a nested type as well. Maybe we should update the general type test to support a nested case rather than adding another one though?